quint/src/errorReporter.ts

```diff
@@ -39,29 +39,42 @@ export function formatError(
     return `error: ${message.explanation}`
   }
 
-  return message.locs.reduce((output, loc) => {
-    // If lineOffset is a number, print the source location.
-    // If lineOfsset is undefined, omit the source location (e.g., in REPL).
-    const locString = lineOffset.isJust()
-      ? `${loc.source}:${loc.start.line + lineOffset.value}:${loc.start.col + 1} - `
-      : ''
-    output += `${locString}error: ${message.explanation}\n`
+  try {
+    // try to extract the location of the error
+    return message.locs.reduce((output, loc) => {
+      // If lineOffset is a number, print the source location.
+      // If lineOfsset is undefined, omit the source location (e.g., in REPL).
+      const locString = lineOffset.isJust()
+        ? `${loc.source}:${loc.start.line + lineOffset.value}:${loc.start.col + 1} - `
+        : ''
+      output += `${locString}error: ${message.explanation}\n`
 
-    const endLine = loc.end ? loc.end.line : loc.start.line
-    const endCol = loc.end ? loc.end.col : loc.start.col
-    for (let i = loc.start.line; i <= endLine; i++) {
-      // finder's indexes start at 1
-      const lineStartIndex = finder.toIndex(i + 1, 1)
-      const line = text.slice(lineStartIndex).split('\n')[0]
+      const endLine = loc.end ? loc.end.line : loc.start.line
+      const endCol = loc.end ? loc.end.col : loc.start.col
+      for (let i = loc.start.line; i <= endLine; i++) {
+        // finder's indexes start at 1
+        const lineStartIndex = finder.toIndex(i + 1, 1)
+        const line = text.slice(lineStartIndex).split('\n')[0]
 
-      const lineStartCol = i === loc.start.line ? loc.start.col : 0
-      const lineEndCol = i === endLine ? endCol : line.length - 1
+        const lineStartCol = i === loc.start.line ? loc.start.col : 0
+        const lineEndCol = i === endLine ? endCol : line.length - 1
 
-      const lineIndex = lineOffset.map(offs => offs + i)
-      output += formatLine(lineIndex, lineStartCol, lineEndCol, line)
-    }
-    return output
-  }, '')
+        const lineIndex = lineOffset.map(offs => offs + i)
+        output += formatLine(lineIndex, lineStartCol, lineEndCol, line)
+      }
+      return output
+    }, '')
+  } catch (error: any) {
+    // If it happens that an error occurs during the error reconstruction,
+    // print that we could not construct a good explanation, instead of throwing.
+    console.error('\nSomething unexpected happened during error reconstruction. Report a bug.\n')
+    console.error('Please include this in the bug report: {')
```

Comment:
Let's be polite first, then use the imperative 😄 

```suggestion
    console.error('\nSomething unexpected happened during error reconstruction. Please report a bug.\n')
    console.error('Include this in the bug report: {')
```